{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Installing package into 'C:/Users/Sophia Schmidt/Documents/R/win-library/3.6'\n",
      "(as 'lib' is unspecified)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "package 'BiocManager' successfully unpacked and MD5 sums checked\n",
      "\n",
      "The downloaded binary packages are in\n",
      "\tC:\\Users\\Sophia Schmidt\\AppData\\Local\\Temp\\RtmpysijdB\\downloaded_packages\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bioconductor version 3.9 (BiocManager 1.30.9), ?BiocManager::install for help\n",
      "Bioconductor version 3.9 (BiocManager 1.30.9), R 3.6.1 (2019-07-05)\n",
      "Installing package(s) 'DESeq2'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "package 'DESeq2' successfully unpacked and MD5 sums checked\n",
      "\n",
      "The downloaded binary packages are in\n",
      "\tC:\\Users\\Sophia Schmidt\\AppData\\Local\\Temp\\RtmpysijdB\\downloaded_packages\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Old packages: 'digest', 'rlang', 'cluster', 'data.table', 'dplyr', 'foreign',\n",
      "  'mgcv', 'purrr'\n",
      "Loading required package: S4Vectors\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "The following objects are masked from 'package:parallel':\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Attaching package: 'IRanges'\n",
      "\n",
      "The following object is masked from 'package:grDevices':\n",
      "\n",
      "    windows\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "Loading required package: SummarizedExperiment\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Loading required package: DelayedArray\n",
      "Loading required package: matrixStats\n",
      "\n",
      "Attaching package: 'matrixStats'\n",
      "\n",
      "The following objects are masked from 'package:Biobase':\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "Loading required package: BiocParallel\n",
      "\n",
      "Attaching package: 'DelayedArray'\n",
      "\n",
      "The following objects are masked from 'package:matrixStats':\n",
      "\n",
      "    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    aperm, apply, rowsum\n",
      "\n"
     ]
    }
   ],
   "source": [
    "install.packages(\"BiocManager\")\n",
    "library(BiocManager)\n",
    "BiocManager::install(\"DESeq2\")\n",
    "library(DESeq2) \n",
    "library(magrittr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error: 'version' is not an exported object from 'namespace:DESeq2'\n",
     "output_type": "error",
     "traceback": [
      "Error: 'version' is not an exported object from 'namespace:DESeq2'\nTraceback:\n",
      "1. DESeq2::version",
      "2. getExportedValue(pkg, name)",
      "3. stop(gettextf(\"'%s' is not an exported object from 'namespace:%s'\", \n .     name, getNamespaceName(ns)), call. = FALSE, domain = NA)"
     ]
    }
   ],
   "source": [
    "DESeq2::version()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "avail <- BiocManager::available()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "18201"
      ],
      "text/latex": [
       "18201"
      ],
      "text/markdown": [
       "18201"
      ],
      "text/plain": [
       "[1] 18201"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(avail) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 5 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ctl1</th><th scope=col>ctl2</th><th scope=col>ctl3</th><th scope=col>ctl4</th><th scope=col>treat1</th><th scope=col>treat2</th><th scope=col>treat3</th><th scope=col>treat4</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ENSMUSG00000102693</th><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000064842</th><td>0</td><td>0</td><td>0</td><td>0</td><td> 1</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000051951</th><td>2</td><td>0</td><td>1</td><td>1</td><td>11</td><td>2</td><td>1</td><td>24</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000102851</th><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000103377</th><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 5 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & ctl1 & ctl2 & ctl3 & ctl4 & treat1 & treat2 & treat3 & treat4\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\tENSMUSG00000102693 & 0 & 0 & 0 & 0 &  0 & 0 & 0 &  0\\\\\n",
       "\tENSMUSG00000064842 & 0 & 0 & 0 & 0 &  1 & 0 & 0 &  0\\\\\n",
       "\tENSMUSG00000051951 & 2 & 0 & 1 & 1 & 11 & 2 & 1 & 24\\\\\n",
       "\tENSMUSG00000102851 & 0 & 0 & 0 & 0 &  0 & 0 & 0 &  0\\\\\n",
       "\tENSMUSG00000103377 & 0 & 0 & 0 & 0 &  0 & 0 & 0 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 5 × 8\n",
       "\n",
       "| <!--/--> | ctl1 &lt;int&gt; | ctl2 &lt;int&gt; | ctl3 &lt;int&gt; | ctl4 &lt;int&gt; | treat1 &lt;int&gt; | treat2 &lt;int&gt; | treat3 &lt;int&gt; | treat4 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| ENSMUSG00000102693 | 0 | 0 | 0 | 0 |  0 | 0 | 0 |  0 |\n",
       "| ENSMUSG00000064842 | 0 | 0 | 0 | 0 |  1 | 0 | 0 |  0 |\n",
       "| ENSMUSG00000051951 | 2 | 0 | 1 | 1 | 11 | 2 | 1 | 24 |\n",
       "| ENSMUSG00000102851 | 0 | 0 | 0 | 0 |  0 | 0 | 0 |  0 |\n",
       "| ENSMUSG00000103377 | 0 | 0 | 0 | 0 |  0 | 0 | 0 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "                   ctl1 ctl2 ctl3 ctl4 treat1 treat2 treat3 treat4\n",
       "ENSMUSG00000102693 0    0    0    0     0     0      0       0    \n",
       "ENSMUSG00000064842 0    0    0    0     1     0      0       0    \n",
       "ENSMUSG00000051951 2    0    1    1    11     2      1      24    \n",
       "ENSMUSG00000102851 0    0    0    0     0     0      0       0    \n",
       "ENSMUSG00000103377 0    0    0    0     0     0      0       0    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "read_counts=read.table(\"C:\\\\Users\\\\Sophia Schmidt\\\\Documents\\\\Uni\\\\Master\\\\Programming for Life Science\\\\raw_countstdl.txt\")\n",
    "head(read_counts, n=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "row.names(read_counts) <- read_counts$Geneid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 5 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>treat3</th><th scope=col>treat4</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ENSMUSG00000102693</th><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000064842</th><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000051951</th><td>1</td><td>24</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000102851</th><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000103377</th><td>0</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 5 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & treat3 & treat4\\\\\n",
       "  & <int> & <int>\\\\\n",
       "\\hline\n",
       "\tENSMUSG00000102693 & 0 &  0\\\\\n",
       "\tENSMUSG00000064842 & 0 &  0\\\\\n",
       "\tENSMUSG00000051951 & 1 & 24\\\\\n",
       "\tENSMUSG00000102851 & 0 &  0\\\\\n",
       "\tENSMUSG00000103377 & 0 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 5 × 2\n",
       "\n",
       "| <!--/--> | treat3 &lt;int&gt; | treat4 &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| ENSMUSG00000102693 | 0 |  0 |\n",
       "| ENSMUSG00000064842 | 0 |  0 |\n",
       "| ENSMUSG00000051951 | 1 | 24 |\n",
       "| ENSMUSG00000102851 | 0 |  0 |\n",
       "| ENSMUSG00000103377 | 0 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "                   treat3 treat4\n",
       "ENSMUSG00000102693 0       0    \n",
       "ENSMUSG00000064842 0       0    \n",
       "ENSMUSG00000051951 1      24    \n",
       "ENSMUSG00000102851 0       0    \n",
       "ENSMUSG00000103377 0       0    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# I did not get that one (?)\n",
    "#read_counts <- read_counts[, -c(1:6)]\n",
    "#head(read_counts, n=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 8 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>condition</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ctl1</th><td>control  </td></tr>\n",
       "\t<tr><th scope=row>ctl2</th><td>control  </td></tr>\n",
       "\t<tr><th scope=row>ctl3</th><td>control  </td></tr>\n",
       "\t<tr><th scope=row>ctl4</th><td>control  </td></tr>\n",
       "\t<tr><th scope=row>treat1</th><td>treatment</td></tr>\n",
       "\t<tr><th scope=row>treat2</th><td>treatment</td></tr>\n",
       "\t<tr><th scope=row>treat3</th><td>treatment</td></tr>\n",
       "\t<tr><th scope=row>treat4</th><td>treatment</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 8 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & condition\\\\\n",
       "  & <fct>\\\\\n",
       "\\hline\n",
       "\tctl1 & control  \\\\\n",
       "\tctl2 & control  \\\\\n",
       "\tctl3 & control  \\\\\n",
       "\tctl4 & control  \\\\\n",
       "\ttreat1 & treatment\\\\\n",
       "\ttreat2 & treatment\\\\\n",
       "\ttreat3 & treatment\\\\\n",
       "\ttreat4 & treatment\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 8 × 1\n",
       "\n",
       "| <!--/--> | condition &lt;fct&gt; |\n",
       "|---|---|\n",
       "| ctl1 | control   |\n",
       "| ctl2 | control   |\n",
       "| ctl3 | control   |\n",
       "| ctl4 | control   |\n",
       "| treat1 | treatment |\n",
       "| treat2 | treatment |\n",
       "| treat3 | treatment |\n",
       "| treat4 | treatment |\n",
       "\n"
      ],
      "text/plain": [
       "       condition\n",
       "ctl1   control  \n",
       "ctl2   control  \n",
       "ctl3   control  \n",
       "ctl4   control  \n",
       "treat1 treatment\n",
       "treat2 treatment\n",
       "treat3 treatment\n",
       "treat4 treatment"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sample_info <- data.frame(condition=rep(c(\"control\",\"treatment\"), each=4),row.names=names(read_counts))\n",
    "sample_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "class: DESeqDataSet \n",
       "dim: 6 8 \n",
       "metadata(1): version\n",
       "assays(1): counts\n",
       "rownames(6): ENSMUSG00000102693 ENSMUSG00000064842 ...\n",
       "  ENSMUSG00000103377 ENSMUSG00000104017\n",
       "rowData names(0):\n",
       "colnames(8): ctl1 ctl2 ... treat3 treat4\n",
       "colData names(1): condition"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "DESeq.ds <- DESeqDataSetFromMatrix(countData = read_counts,\n",
    "                                      colData = sample_info,\n",
    "                                      design = ~condition)\n",
    "head(DESeq.ds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataFrame with 6 rows and 1 column\n",
       "       condition\n",
       "        <factor>\n",
       "ctl1     control\n",
       "ctl2     control\n",
       "ctl3     control\n",
       "ctl4     control\n",
       "treat1 treatment\n",
       "treat2 treatment"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 6 × 8 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ctl1</th><th scope=col>ctl2</th><th scope=col>ctl3</th><th scope=col>ctl4</th><th scope=col>treat1</th><th scope=col>treat2</th><th scope=col>treat3</th><th scope=col>treat4</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ENSMUSG00000102693</th><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000064842</th><td>0</td><td>0</td><td>0</td><td>0</td><td> 1</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000051951</th><td>2</td><td>0</td><td>1</td><td>1</td><td>11</td><td>2</td><td>1</td><td>24</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000102851</th><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000103377</th><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>ENSMUSG00000104017</th><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 8 of type int\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & ctl1 & ctl2 & ctl3 & ctl4 & treat1 & treat2 & treat3 & treat4\\\\\n",
       "\\hline\n",
       "\tENSMUSG00000102693 & 0 & 0 & 0 & 0 &  0 & 0 & 0 &  0\\\\\n",
       "\tENSMUSG00000064842 & 0 & 0 & 0 & 0 &  1 & 0 & 0 &  0\\\\\n",
       "\tENSMUSG00000051951 & 2 & 0 & 1 & 1 & 11 & 2 & 1 & 24\\\\\n",
       "\tENSMUSG00000102851 & 0 & 0 & 0 & 0 &  0 & 0 & 0 &  0\\\\\n",
       "\tENSMUSG00000103377 & 0 & 0 & 0 & 0 &  0 & 0 & 0 &  0\\\\\n",
       "\tENSMUSG00000104017 & 0 & 0 & 0 & 0 &  0 & 0 & 0 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 8 of type int\n",
       "\n",
       "| <!--/--> | ctl1 | ctl2 | ctl3 | ctl4 | treat1 | treat2 | treat3 | treat4 |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| ENSMUSG00000102693 | 0 | 0 | 0 | 0 |  0 | 0 | 0 |  0 |\n",
       "| ENSMUSG00000064842 | 0 | 0 | 0 | 0 |  1 | 0 | 0 |  0 |\n",
       "| ENSMUSG00000051951 | 2 | 0 | 1 | 1 | 11 | 2 | 1 | 24 |\n",
       "| ENSMUSG00000102851 | 0 | 0 | 0 | 0 |  0 | 0 | 0 |  0 |\n",
       "| ENSMUSG00000103377 | 0 | 0 | 0 | 0 |  0 | 0 | 0 |  0 |\n",
       "| ENSMUSG00000104017 | 0 | 0 | 0 | 0 |  0 | 0 | 0 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "                   ctl1 ctl2 ctl3 ctl4 treat1 treat2 treat3 treat4\n",
       "ENSMUSG00000102693 0    0    0    0     0     0      0       0    \n",
       "ENSMUSG00000064842 0    0    0    0     1     0      0       0    \n",
       "ENSMUSG00000051951 2    0    1    1    11     2      1      24    \n",
       "ENSMUSG00000102851 0    0    0    0     0     0      0       0    \n",
       "ENSMUSG00000103377 0    0    0    0     0     0      0       0    \n",
       "ENSMUSG00000104017 0    0    0    0     0     0      0       0    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "GRangesList object of length 6:\n",
       "$ENSMUSG00000102693 \n",
       "GRanges object with 0 ranges and 0 metadata columns:\n",
       "   seqnames    ranges strand\n",
       "      <Rle> <IRanges>  <Rle>\n",
       "\n",
       "$ENSMUSG00000064842 \n",
       "GRanges object with 0 ranges and 0 metadata columns:\n",
       "     seqnames ranges strand\n",
       "\n",
       "$ENSMUSG00000051951 \n",
       "GRanges object with 0 ranges and 0 metadata columns:\n",
       "     seqnames ranges strand\n",
       "\n",
       "...\n",
       "<3 more elements>\n",
       "-------\n",
       "seqinfo: no sequences"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Check the DESeqDataSet\n",
    "colData(DESeq.ds) %>% head\n",
    "assay(DESeq.ds) %>% head\n",
    "rowRanges(DESeq.ds) %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove the genes without any counts\n",
    "DESeq.ds <- DESeq.ds[rowSums(counts(DESeq.ds)) > 0,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl class=dl-horizontal>\n",
       "\t<dt>ctl1</dt>\n",
       "\t\t<dd>0.861385212241083</dd>\n",
       "\t<dt>ctl2</dt>\n",
       "\t\t<dd>0.760175549092362</dd>\n",
       "\t<dt>ctl3</dt>\n",
       "\t\t<dd>0.808985675298092</dd>\n",
       "\t<dt>ctl4</dt>\n",
       "\t\t<dd>0.833769556412077</dd>\n",
       "\t<dt>treat1</dt>\n",
       "\t\t<dd>0.94871557100813</dd>\n",
       "\t<dt>treat2</dt>\n",
       "\t\t<dd>0.819283353107582</dd>\n",
       "\t<dt>treat3</dt>\n",
       "\t\t<dd>0.978758960142215</dd>\n",
       "\t<dt>treat4</dt>\n",
       "\t\t<dd>4.05397505356715</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[ctl1] 0.861385212241083\n",
       "\\item[ctl2] 0.760175549092362\n",
       "\\item[ctl3] 0.808985675298092\n",
       "\\item[ctl4] 0.833769556412077\n",
       "\\item[treat1] 0.94871557100813\n",
       "\\item[treat2] 0.819283353107582\n",
       "\\item[treat3] 0.978758960142215\n",
       "\\item[treat4] 4.05397505356715\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "ctl1\n",
       ":   0.861385212241083ctl2\n",
       ":   0.760175549092362ctl3\n",
       ":   0.808985675298092ctl4\n",
       ":   0.833769556412077treat1\n",
       ":   0.94871557100813treat2\n",
       ":   0.819283353107582treat3\n",
       ":   0.978758960142215treat4\n",
       ":   4.05397505356715\n",
       "\n"
      ],
      "text/plain": [
       "     ctl1      ctl2      ctl3      ctl4    treat1    treat2    treat3    treat4 \n",
       "0.8613852 0.7601755 0.8089857 0.8337696 0.9487156 0.8192834 0.9787590 4.0539751 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Normalize the count reads with size factor\n",
    "DESeq.ds <- estimateSizeFactors(DESeq.ds)\n",
    "sizeFactors(DESeq.ds)\n",
    "norm_counts <- counts(DESeq.ds, normalized = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Transform normalized read counts to log2 scale using a pseudocount of 1\n",
    "log.norm.counts <- log2(norm_counts + 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAAMFBMVEUAAABNTU1oaGh8fHyM\njIyampqnp6eysrK9vb3Hx8fQ0NDZ2dnh4eHp6enw8PD////QFLu4AAAACXBIWXMAABJ0AAAS\ndAHeZh94AAAgAElEQVR4nO2d22KiQBBEQY1rvP7/3654ixA1MFMjRXvOw24u3dUzTZcKGq0O\nAJBNNfYCACKAkQAEYCQAARgJQABGAhCAkQAEYCQAARgJQABGAhCAkQAEYCQAARgJQABGAhCA\nkQAEYCQAARgJQABGAhCAkQAEYCQAARgJQABGAhCAkQAEYCQAARgJQABGAhCAkQAEYCQAARgJ\nQABGAhCAkQAEYCQAARgJQABGAhCAkQAEYCQAARgJQABGAhCAkQAEYCQAARgJQABGAhCAkX5T\nVX26slvWVbVYd376lVr031EtOfmefqtXJv5Cso+JgZF+02ui1tWZ5f0PN3VqP/81WjGMlN6E\nKfOJe/6LPhO1r66sBmY+ZlZV28TUDqMbSXfPNiU+cc9/0WcSjvcg8+1h/1VVs4GZ6TULK2Gk\nHD5xz39xm4T1V/N463oatDt+N19dfjuvqk0r9vx1df7++O9udnrU9704fj1b7q6x62Pi1/m+\nZ//v+HW1+G7XbBe9CTW/X82q2bHoqq7mm/Ov98fztPos3qyvrmarzhz/rKQVe7+sx4nH+OO9\n5Py6+e6q7v9v7eunCa39hQcj/eY6J/PLUCxO323O38zPv10sWrN0/frHSM0U/kicXNecUZ2/\nayZuV19+Nb/P7BS9CV0qH22xvOndJDa/13e3prNAK7a1rMeJ1/jlg1b8NtLPvm5bud/fB4CR\nfnOZj8V12s7jU9++vevZ5jpbl7wfIx35PqyOY7Q/HJbnqB+B5qrCVxNw2M+bs6yfzE7Rq9BP\nbv1gSfXT9f0I3Me2l/U48fbT9cNV3Teqta+b0v3+PgCM9JvzfDSX5Vb74wOU8zB9H2ew+a9u\nzdvsPGjtzPP9x/78693Pz6uzxNftu+Z3+/NZ1iWzU/QmdPni6IBqtj39d7j6YX+e2Mv61u31\n3QRasa1lPU48xtfbkw1mD1d1v932vu5+ere/8GCk35wn4et6U7o83dIuLpb5vp+3eeeBy92E\nrX///PLT/fm7ZnBv51/XzE7RH6Hq9uhwc4s+LulissXP+ta/jHT6cSu2VfRx4uWn+9m/3cNV\n3S+6va/rT9v7Cw9G+s1t7E+jd9idvq+vY3Y3b0cf1fvD3cOZuwnbX2J238t5dVO8U/h3eTS0\nvlftFP35/tet/U/Z8+O16sH67gRasa1lPU7sftNd1YNFdb5t7y88GOk37Wm5fPV73nbN457r\nQ6RfRjrHfM+q2y+7v11eJ3t3+DWTvxbx2khP/XBf824l98vqZ6SHq3pppPb+woORftO5M9id\nbsV/3SPt69vjuqdGah4Hzr5W24fzd9h/ny+GzQ/3Q3hf9LWR6ofD/thIrdjWsgbeI92v6rWR\nWvsLD0b6TXU9A3l5jjTvvDzoJ/Pn/+u1iMdGalh/tX63eHI28tBIi/szsYfncD8CrdjWsh4n\nzu/PkR6sqjHW5g8j3e8vPB+xyYGcD/2Tq3aXB0fNCfjvW9rmd/vDr9F6eI80u5293N3IP7s+\n9tBIzZI2p//m58tsv68q3t83/sS2lvU48eVVu/p0G7Kpnxtp391feDDSby4DcXvW8vxi0vbT\nLdvWd1eamOWPwuVe63pluT1wx+mc707n5J2M+6IvjfSzpM2v9XX20o5tL+uP55FWv1f11c7o\nGOnShPb+woORftMd6vOLsi8v9768AKAzSxean96d8VxfNNA8i7p5erFhfl+zU/S1kVqvQL/U\nWjwx0n1se1mPEzcPX9lwXtXu8qvHRro2obW/8GCk3/wM31d9d/12+3V67dmrm/HmbKJ1atPk\n1F/b3enpm87Anc8f5qt2zXbR10Y6vx7u+jdRrdcC/tpLK7a1rMeJp9fm/fy91YNWfD+52HBt\nQmt/4cFIA9l/yGN+GAZG6kl1enHNYTtvvTYA4AxG6snP+Xb1Ic/VwxAwUk9ufxXwGRehYCAY\nqS/7f811rfpTXjsGw8BIAAIwEoAAjAQgACMBCMBIAAIwEoAAjAQgACMBCMBIAAIwEoAAjAQg\nACMBCMBIAAIwEoAAjAQgACMBCMBIAAIwEoAAjAQgACMBCMBIAAIwEoAAjAQgACMBCMBIAAIw\nEoAAjAQgACMBCMBIAAIwEoAAjAQgACMBCMBIAAIwEoCANxipApgYCVOuN84IJQCUYCQAARgJ\nQABGAhCAkQAEYCQAARgJQABGAhCAkQAEYCQAARgJQABGAhCAkQAEYCQAARgJQABGAhCAkQAE\nYCQAARgJQABGAhCAkQAEYCQAARjpLRR4MydrPm2/GGkkwm+wQ/z9YqTHCyh8izr6Bt9M/P0G\nMdLUBj/+YLWJv98gRpILTEzPnfj7xUij6EE0MNIoehANjDRJPu/ysjtBjeTOp91juq8vn7ca\nafNvcbq5XCw3pUpMBIwUjTcaaT+7e+gxL1KiGO6Db9ewDu7ry+eNRlpW9ff29NVuXVfLEiV0\nAh+mN7Xn4XLR7/eNRqqr7e3rbVWXKKETmJieGrf9+hv9jUZq7fb11sMbyR33/fodX+6RRtFz\nx32/fsf3vedI693pq+mdI7njN1hl8dvvOy9/z+8e0872RUpMBb9BKEv8/b73eaTl6XmkevGP\n55G89dS4ry8fXtnQC/fBt2tYB/f15eNjJOlLxdwHFT0vpnWOtP+qqvn6IvLZl7/jD9an6b3z\nJUL1+YV2Z5HPNpI77vv1O75vvfy9OrppVZ9eZoeRrHHfr9/xfesTsqf/dvVsNzkjueM3WGXx\n2+8ILxHaz+fFjeSO3yCUJf5+32ikWXV9EnY2x0jeemrc15fPG420qr4uX+2q+cSM5D74dg3r\n4L6+fN55+Xt5c8/6j6eK7M6R0PPSUzOpc6TDYbu4frX7wkjOuO/XT8/nlQ3SEn6Nnhbu+/U7\nvhhpFD133Pfrd3wxUgj8BqssfvsNaiR3/AahLPH3i5FGIf5gtXFfXz4YqRfug2/XsA7u68sn\nqJHcBxU9LzhHKiUwMT017vv108NIo+i5475fv+OLkUbRc8d9v37HFyOFwG+wyuK336BGcsdv\nEMoSf78YaRTiD1Yb9/Xlg5F64T74dg3r4L6+fIIayX1Q0fOCc6RSAhPTU+O+Xz89jDSKnjvu\n+/U7vhhpFD133Pfrd3wxUgj8BqssfvsNaiR3/AahLPH3i5FGIf5gtXFfXz4YqRfug2/XsA7u\n68snqJHcBxU9LzhHKiUwMT017vv108NIo+i5475fv+OLkUbRc8d9v37HFyOFwG+wyuK336BG\ncsdvEMoSf78YaRTiD1Yb9/Xlg5F64T74dg3r4L6+fIIayX1Q0fOCc6RSAhPTU+O+Xz89jDSK\nnjvu+/U7vhhpFD133Pfrd3wxUgj8BqssfvsNaiR3/AahLPH3i5FGIf5gtXFfXz4YqRfug2/X\nsA7u68snqJHcBxU9LzhHKiUwMT017vv108NIo+i5475fv+OLkUbRc8d9v37HFyOFwG+wyuK3\n36BGcsdvEMoSf78YaRTiD1Yb9/Xlg5F64T74dg3r4L6+fIIayX1Q0fOCc6RSAhPTU+O+Xz89\njDSKnjvu+/U7vhhpFD133Pfrd3wxUgj8BqssfvsNaiR3/AahLPH3i5FGIf5gtXFfXz4YqRfu\ng2/XsA7u68snqJHcBxU9LzhHKiUwMT017vv108NIo+i5475fv+OLkUbRc8d9v37HFyOFwG+w\nyuK336BGcsdvEMoSf78YaRTiD1Yb9/Xlg5F64T74dg3r4L6+fIIayX1Q0fOCc6RSAhPTU+O+\nXz89jDSKnjvu+/U7vhhpFD133Pfrd3wxUgj8BqssfvsNaiR3/AahLPH3i5FGIf5gtXFfXz4Y\nqRfug2/XsA7u68snqJHcBxU9LzhHKiUwMT017vv108NIo+i5475fv+OLkUbRc8d9v37HFyOF\nwG+wyuK336BGcsdvEMoSf78YaRTiD1Yb9/Xlg5F64T74dg3r4L6+fIIayX1Q0fOCc6RSAhPT\nU+O+Xz89jDSKnjvu+/U7vhhpFD133Pfrd3wxUgj8BqssfvsNaiR3/AahLPH3i5FGIf5gtXFf\nXz4YqRfug2/XsA7u68snqJHcBxU9LzhHKiUwMT017vv108NIo+i5475fv+OLkUbRc8d9v37H\nFyOFwG+wyuK336BGcsdvEMoSf78YaRTiD1Yb9/Xlg5F64T74dg3r4L6+fIIayX1Q0fOCc6RS\nAhPTU+O+Xz89jDSKnjvu+/U7vhhpFD133Pfrd3wxUgj8BqssfvsNaiR3/AahLPH3i5FGIf5g\ntXFfXz4YqRfug2/XsA7u68snqJHcBxU9LzhHKiUwMT017vv108NIo+i5475fv+OLkUbRc8d9\nv37HFyOFwG+wyuK336BGcsdvEMoSf78YaRTiD1Yb9/Xlg5F64T74dg3r4L6+fIIayX1Q0fOC\nc6RSAhPTU+O+Xz89jDSKnjvu+/U7vhhpFD133Pfrd3wxUgj8BqssfvsNaiR3/AahLPH3i5FG\nIf5gtXFfXz4YqRfug2/XsA7u68vn/UZazapqsS5awn9Q0fNiUudI1SlxXp1YFimhE5iYnhr3\n/frpvdtIy2q5Pxx2y2pVooROoLCeO+779Tu+7zZSXe2br/fVrEQJnUBhPXfc9+t3fN9tpKq6\n+0ZeQicwMfwGqyx++323kb6uRqpLlJgMfoNQlvj7fauRFv9W6+r7+OV++fpqg1+fxMQfrDbu\n68vnrUY6c/qy3pcoUQz3wbdrWAf39eXzzueRttvVarE4XXJYvvSR3zkSel56aiZ1jvTOEu6D\nEH+wPk3Px0jVPdliihUV1HPHfb9+x9fHSNISfo2eFu779Tu+GCkEfoNVFr/9BjWSO36DUJb4\n+x3h8neP0yC/PomJP1ht3NeXzxuNtJqwkdwH365hHdzXl89bn0eq56VLyATQs9ZTM7FzpO0f\nf4YkKKESmJieGvf9+um992LDqtqWLiESKKznjvt+/Y5v0Kt2fo2eFu779Tu+GCkEfoNVFr/9\nBjWSO36DUJb4+8VIoxB/sNq4ry8fjNQL98G3a1gH9/XlE9RI7oOKnhecI5USmJieGvf9+ulh\npFH03HHfr9/xxUij6Lnjvl+/44uRQuA3WGXx229QI7njNwhlib9fjDQK8Qerjfv68sFIvXAf\nfLuGdXBfXz5BjeQ+qOh5wTlSKYGJ6alx36+fHkYaRc8d9/36HV+MNIqeO+779Tu+GCkEfoNV\nFr/9BjWSO36DUJb4+8VIoxB/sNq4ry8fjNQL98G3a1gH9/XlE9RI7oOKnhecI5USmJieGvf9\n+ulhpFH03HHfr9/xxUij6Lnjvl+/44uRQuA3WGXx229QI7njNwhlib9fjDQK8Qerjfv68sFI\nvXAffLuGdXBfXz5BjeQ+qOh5wTlSKYGJ6alx36+fHkYaRc8d9/36HV+MNIqeO+779Tu+GCkE\nfoNVFr/9BjWSO36DUJb4+8VIoxB/sNq4ry8fjNQL98G3a1gH9/XlE9RI7oOKnhecI5USmJie\nGvf9+ulhpFH03HHfr9/xxUij6Lnjvl+/45trpM1yXlXVfLnJXsnTEqMITAy/wSqL337zjPQ9\nq67M1tlreVgiJn6DUJb4+80x0m5ezVfb/fGr/ebf8evdmKuaFvEHq437+vLJMNK6Wu7vfrxb\nVrI7Jbu+uw++XcM6uK8vnwwjLfadX+y/clfTLTGeAHrWemrGPkcqBUYaGff9+ulhpFH03HHf\nr9/xzTXSanY8O5pVs+CXv90HS437fv2Ob6aR1tXxu7q5/C11kp2R3PEbrLL47TfTSPPq+7Ct\nZofvap69lCclQuI3CGWJv99MIzV3SNtqef5Ch1+fxMQfrDbu68tHYKRF8/xRcCO5D75dwzq4\nry+f7Id223VVH+we2rkPKnpejH6OtG6uM/xr7pCUL7XDSGPjvl8/vezL33VzhnSYfWev5GmJ\nUQQK67njvl+/48sTsqPoueO+X7/ji5FC4DdYZfHbr+Cq3Ym6zl7KkxIh8RuEssTfr8hIu+CX\nv9XEH6w27uvLJ8NI6+qe2cirKov74Ns1rIP7+vLJuUea3fso9mvt0PPSU+NzjqQFI42M+379\n9LhqN4qeO+779Tu+GGkUPXfc9+t3fHON9O92opS9lGclRhGYGH6DVRa//WYa6d/P5YbspTwp\nERK/QShL/P1mGqmuVrKlPCkRkviD1cZ9ffkEvWqnxn3w7RrWwX19+WQaaVF139xOgt05Enpe\nempGP0fa1XPt+wf9LjGOwMT01Ljv108v+6Gd58UGv0ZPC/f9+h1fjDSKnjvu+/U7vjwhGwK/\nwSqL336DGskdv0EoS/z9YqRRiD9YbdzXl0/QcyQ17oNv17AO7uvLJ6iR3AcVPS9czpE280X2\nSv4o8WaBiempcd+vn57oHGlfyT6t71mJ9woU1nPHfb9+x1d1sYGHdqFw36/f8RUZaVV5vR2X\n+yCo8RussvjtV3ax4V/2Up6UCInfIJQl/n5FRppp/yzJr09i4g9WG/f15cMTsr1wH3y7hnVw\nX18+QY3kPqjoeTH6OdLh8D0/PrJbaD/VBSONjft+/fRyjTS/nCRJP7AvvpHccd+v3/HNNNKq\nqpuP6luL3wQFI42M+379jm+mkWbV9vT/1uxN9N0HQY3fYJXFb7/Zl7+7X0hwP47Z+A1CWeLv\nV3aP5PXKBnfiD1Yb9/XlE/QcSY374Ns1rIP7+vLhqh16E9RTM/o50uHwveB5pPH11Ljv10+P\nVzaMoueO+379ji9GGkXPHff9+h3fXCPtl83lunqpfQtwOyO54zdYZfHbb6aRdvXpCaSqqnfZ\nS3lSIiR+g1CW+PvNNNK8+mrui/bLSvruJ359EhN/sNq4ry8fXtnQC/fBt2tYB/f15ZNppPry\n+Uh7MyO5Dyp6Xox+jrSsTp+PtJlXy+ylPCkxjsDE9NS479dPj1c2jKLnjvt+/Y6v5pUNc/FH\nMmOkkXHfr9/x5QnZEPgNVln89qs0ku6Cg/txzMZvEMoSf78YaRTiD1Yb9/Xlg5F64T74dg3r\n4L6+fIIayX1Q0fOCc6RSAhPTU+O+Xz89jDSKnjvu+/U7vhhpFD133Pfrd3wxUgj8BqssfvsN\naiR3/AahLPH3i5FGIf5gtXFfXz4YqRfug2/XsA7u68uH19qhN0E9NaOeI1VtspeStyqxwMT0\n1Ljv108PI42i5477fv2Ob+5Du8Xpvb839VefzM2/xclzi+VGviqxQGE9d9z363d8M420vH0a\nxd9/ar6f3d1/vf6LWjsjueM3WGXx22+mkYa8i9Cyqr/Pttut69fGcz+O2fgNQlni7zfTSPWA\nz0e6xvaI9+uTmPiD1cZ9fflkP7Srm9Od4z3Mv7/zHt6VqVZVFvfBt2tYB/f15ZN7seH6LkI9\n3mj1nfdI7oOKnhejnyNdPx9p3SPveO+1Pr9DePlzJPdBiD9Yn6b3zlc2zO+u2s1efnxFeCO5\n475fv+P71pcIbZan55HqxT+eR/LGfb9+x1dlpI3Xp1G4D4Iav8Eqi99+c420lL1EqNjrjRzx\nG4SyxN9v9uXvK30uNySVCEn8wWrjvr58sp+Q/T7Mq91uXv1x1pNewgL3wbdrWAf39eUjeInQ\nv+O90Vb7cRR250joeempGf0cqTHSulr1eq3dgD+7wEgj475fP71MIy2OD+121eyw6WGkFUaa\nDO779Tu+mUZaN4Y4PdHa4w+StnXfx38YaWTc9+t3fHMvf/9rvvuq+n3yZZ+/WvpdIgX3QVDj\nN1hl8dvvW1/ZcHx0t/07KK/ENPAbhLLE3+97jSQrUeVSfg+v12+uN7j+xI9HPtlGWi+aLix2\novU8KpESIM4fe/DdB/Xdx8OPXCPNz0epqqVOeruR1IMq1xPvd+z4v/LV/Su9/mwjrar5vln1\nqs9Vu7QSaQED8z9Oz9zo7jccEoX2S4T25+di3/y+du4HDj0vPXW8RKH7ygaMhJ67njpeonCf\nMrvcI22rWfZSnpRICxiYj15sPXW8ROHBOdK6bl5vpwMjoSfVK0/uVbvF5eRT+uJvjISeVq88\nuUY6PY9ULb5Fy3lYIiVgYD56sfXKk22kImAk9KR66niJwn3KouerUDNKpAUMzEcvtp46XqLQ\nvfxdAIyEnlRPHS9R6F7+LgBGQk+qp46XKNyn7Bdz6buePCiRFjAwH73Yeup4iUL7oV3qywR7\nl0gLGJiPXmy98mAk9D5Arzxc/kbvA/TKg5HQ+wQ98Z+N/F1xwCIX3Qt2e9nfJGEk9Kz1UiKe\npqyr5b2Vdkvd+39jJPSs9VIinqfs5tV8tW3MtN/8O36t+2tzjISetV5KxKuU79ntQeRM+XEU\nGAk9a72UiNcpm2Xz9ifzpfZpWYyEnrVeSoQiRV7CvdHoxdZLiVCkyEu4Nxq92HopEc9T9l/H\nB3WXUyNe2YDeB+mlRDxN2denqwznT2HGSOh9kF5KxNOUZfOGJ/vV+bNaMBJ6H6SXEvE0pT5/\nsatnO4yE3kfppUQ8Tbl6Zz+fYyT0PkovJeJpys9fx87mGAm9T9JLiXia8vPG+btqjpHQ+yC9\nlIjnKcube9b8YR96n6SXEvEiZbu4frX7wkjofY5eSoQiRV7CvdHoxdZLiVCkyEu4Nxq92Hop\nES9T7v4Ydy5801WMhJ61XkrEy5TW37XXw7VSV+XeaPRi66VEvE75qptXra7ranNYVLL7JIyE\nnrVeSsTLlGW1Pf2/reaHve5T+zASetZ6KREvU25PH90+TFYCRkLPWi8l4mVKfbtHqjESeh+j\nlxLxMmVZXc+Rlodv3edfYiT0rPVSIl6nzH8+Q7bSfSAzRkLPWi8l4o+U82fINndL1b/hWr1K\npAQMzEcPvSF6KRGKFHkJ90ajF1svJUKRIi/h3mj0YuulRPyR8t2cJS2+h8v0L5ESMDAfPfSG\n6KVEvE65u9ggBCOhZ62XEvEyZXW7/C27YtctkRYwMB899IbopUS8TJndnpCVvTyoWyItYGA+\neugN0UuJeJnSeomQDoyEnrVeSsTLlNn9S4R0YCT0rPVSIl6mcI6E3ifqpUS8TuGqHXofqJcS\n8UfK94LnkdD7NL2UCEWKvIR7o9GLrZcSoUiRl3BvNHqx9VIinqZUbYYLZazKvdHoxdZLiXia\ngpHQ+1S9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK90ejF1kuJ\nUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PR\ni62XEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5\nCfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVS\nIhQp8hLujUYvtl5KhCJFXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o\n9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJF\nXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9\nlAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK90ejF1kuJUKTIS7g3\nGr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PRi62XEqFI\nkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdb\nLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLu\njUYvtl5KhCJFXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQo\nUuQl3BuNXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK90ejF\n1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryE\ne6PRi62XEqFIuWc1q6rFOrOEe6PRi62XEqFIOeedEufViWVeCfdGoxdbLyVCkXLOaxKX1XJ/\nOOyW1SqrhHuj0YutlxKhSDnnNYl1tW++3lezrBLujUYvtl5KhCLlnFdd/7n7//7Xd5Raw7N8\n9NAbopcSoUg55zWJX1cj1Vkl3BuNXmy9lAhFyjmvWvxbravv45f75eurDRgJPWu9lAhFyjnv\n52FbVdX7rBLujUYvtl5KhCLlzHa7Wi0Wp0sOy5c+wkjoeeulRChS5CXcG41ebL2UCEWKvIR7\no9GLrZcSoUiRl3BvNHqx9VIiFCnyEu6NRi+2XkqEIkVewr3R6MXWS4lQpMhLuDcavdh6KRGK\nFHkJ90ajF1svJUKRIi/h3mj0YuulRChS5CXcG41ebL2UCEWKvIR7o9GLrZcSoUiRl3BvNHqx\n9VIiFCnyEu6NRi+2XkqEIkVewr3R6MXWS4lQpMhLuDfaXu/vP0J+pfYr236/Yr2UCEWKvIR7\no931DodDrz/pf5j0SCxlUS/y3fVSIhQp8hIFbqGTJR/N1lQG4WqM6g8OfxjPfb8YqdAaHuXf\nv/FKj8FqxZdeX8lB+Guvr/ZZan3ueikRihR5CQYrT+/0M+cbjj/6+1LrQTJGKrSGj3+oox5U\nsd7t56ZGl1QIaqSXxniq8yTLfvBTxV7kKwf/kW4fzSHrHQJG6pmfMacPs90H1f2h8b1OH81u\n/JD19gEj9cx3vwf59fvMQbU3urh/GKnQGqZ2D6LWc7/hmMLxza6Akf7Wa/2uJy/1P2xQ3fVS\nIhQp8hJTaLT1PQh6WXopEYoUeQn3RnMPElsvJUKRIi/BoKI3SE88L39X7COcsJgSJXjohP3v\nT7IAAAppSURBVF5fvdPPhPPSr6I+pWiJv5rSpyHug4Benl7rd4J5GVZRl1KsxHXLPZrysjnu\ng4Bent7pZw/ubV6ZaJihpmukhFuNH+OlFHypi563XsKs3OeGPUfK6MvDbPtBQM9KLyVCkSIv\n4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5K\nhCJFXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuN\nXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK90ejF1kuJUKTI\nS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PRi62X\nEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5CfdG\noxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp\n8hLujUYvtl5KhCJFXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLr\npUQoUuQl3BuNXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK9\n0ejF1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhF\niryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK90ejF1kuJUKTIS7g3Gr3Y\neikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PRi62XEqFIkZdw\nbzR6sfVSIhQp8hLujUYvtl5KhCJFXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVC\nkSIv4d5o9GLrpUQoUuQl3BuNXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLujUYv\ntl5KhCJFXsK90ejF1kuJUKTIS7g3Gr3YeikRihR5CfdGoxdbLyVCkSIv4d5o9GLrpUQoUuQl\n3BuNXmy9lAhFiryEe6PRi62XEqFIkZdwbzR6sfVSIhQp8hLyRudSen3oWemlRChS5CXKNyYz\n/tOM+WH7TYlQpMhLqA+cO+6DqgYjaZjanE8OjJSnlxKhSDEs8dm436NjJA3ZJdRrRO+9etN7\nKIuR0PsAvfL3wBgJPfQEehgJPfQwUjEB9NArrTAFIwG8F4wEIAAjAQgIaiS/x9DoxdbDSOih\nh5GKCaCHXmkFjIQeegIFjIQeegKFKRgJ4L1gJAABGAlAQFAj+T2GRi+2HkZCDz2MVEwAPfRK\nK2Ak9NATKGAk9NATKEzBSADvBSMBCMBIAALeaqTNv8XpvY0Wy02pEioB9NArrZBadD+7e5+w\neZESOgH00CutkFp0WdXf29NXu3VdLUuU0Amgh15phdSidbW9fb2t6hIldALooVdaIbVo631f\nf78J7LB3iH2VnPSWs+ihl64X9h4J4L289xxpvTt9Vf4cCeC9vPPy9/zurnO2L1ICYBze+zzS\n8vQ8Ur34V/p5JID3wisbAARgJAABGAlAAEYCEICRAARgJAABGAlAAEYCEICRAARgJAABGAlA\nAEYCEICRAARgJAABGAlAAEYCEICRAARgJAABpkYCmBgJU643jhy/NxBE75P0fIsOxL3R6MXW\n8y06EPdGoxdbz7foQNwbjV5sPd+iA3FvNHqx9XyLDsS90ejF1vMtOhD3RqMXW8+36EDcG41e\nbD3fogNxbzR6sfV8iw7EvdHoxdbzLToQ90ajF1vPt+hA3BuNXmw936IA0cBIAAIwEoAAjAQg\nACMBCMBIAAIwEoAAjAQgACMBCMBIAAIwEoAAjAQgACMBCMBIAAIwEoAAjAQgwN1I6+af65ua\nr/JXe6e3mlX1ci/T239V1dc2U66938Nm0IbXA0L6tLK/Xr9W9tbr2coh+x3YyuGYG2l2Wt9l\nsLYpnxLwVG95+tiBOs9Jd3r1SS/TSa39Hvb1kA3P/g6+hfRpZX+9fq3sr9evlUP2O7CVCZgb\nqbobrG2db6QfvW31tW9umL9EestGaVktVOtrWAzacI/ga0ivVvYO6dnK3no9WzloC8NamcB0\njLSq5kojLaq7Hwj06mqfL9c20vewD+rpP1j9Wtk7pGcre+v1bOWQLQxsZQKmRlrW1Xx3OFw+\n9OnUhGqZMaYP9M4kSj7Tq2rd+nbDbjluufvZ6ca8OW9ZnX6zPt4c18ufkH6tHKB3TZDq/dXK\nIXoDW5mCp5Hml8fc94O1zbi9f6R3Yl/NlXrLaqVb37zaJRnpOEbL02OZI83u/p0/hG75M1i9\nWjlA78RfrRyo92crh+gNbGUKlkb6rub7w9e5Fc331yakNuOZXvMgp8eln75636ejp9L7V30P\n2/A1d948MFo3/+3nzfaqo1CztsP9xns/Luqp16OVQ/T6tLK/3uBWJmBppEW1aW7hapWRnukd\ndnXSxYEneqtFXf1T6W2bxytJRtqcJZvx2v+csacbqZdej1YO0evTyt56w1uZgKWRusc710jP\n9PZ10gO7p3qH491KymO7R3qz5mpykpEu31xovtmt/83TjdRLr0crB+n1aGVvveGtTOCjjTSf\nafUudysKva/TAyWNkea3rzRGeqTXo5WD9Hq0sq9eQisT+GAj7WbNhTLl+g6JS3ygV91PxhCV\n22Ddfv5VzVbrXbaRXun1auWg9fVYYl+9hFYmYGmkufgc6aHeOu2C3TO985MfuyrlTu6BXq6R\nFj+n/qef5BrppV6/VvbW69nKvnofbKRVcwlmeb6K1dzS5Rrpkd4u3UcP9U5Px+8XSedIz/Y7\n9KHdXe53VW8b4dNp9uawvZ4z7G7BOr2ereyt17OVg/b7mQ/tbs+rHGbVz+10RjMe6H3l3Ew9\nWl99eyJDo9cwaHXt3POJQr27vhLudHXrHNJTubdez1b2X1+/Vg7a74caqWnGorkx2cwkRnqg\nl3d//2h9y/r44Fy2vhODVtfJXR2n6Ot0g3wc9Plm3dxYn0N6KvfW69nKAevr1cpB+/1UIwFM\nC4wEIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjA\nSAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwE\nIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjASAAC\nMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAj\nAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKAAIwEIAAjAQjASAACMBKA\nAIwEIAAjAQjASAACMBKAAIwEIAAjAQj4DwR732Z99KrYAAAAAElFTkSuQmCC",
      "text/plain": [
       "Plot with title \"log2-transformed read counts\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Generate a boxplot of log2-transformed read counts\n",
    "boxplot(log.norm.counts, notch = TRUE ,\n",
    "             main = \"log2-transformed read counts\",\n",
    "              ylab = \"log2(read_counts)\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
